src/watchdog.rs

```diff
@@ -0,0 +1,185 @@
+#[cfg(esp_idf_esp_task_wdt)]
+pub mod task {
+    //! ## Example
+    //!
+    //! ```rust, ignore
+    //! # fn main() -> Result<()> {
+    //! let mut watchdog = TaskWatchdog::take()?.unwrap();
+    //!
+    //! watchdog.reconfigure(WatchdogConfig {
+    //!     duration: Duration::from_secs(2),
+    //!     panic_on_trigger: true,
+    //!     ..Default::default()
+    //! })?;
+    //!
+    //! let task = Task::current();
+    //!
+    //! task.set_watchdog(&mut watchdog)?;
+    //!
+    //! loop {
+    //!     task.reset_watchdog()?;
+    //!     unsafe { vTaskDelay(1) };
+    //! }
+    //! # }
+    //! ```
+
+    use esp_idf_sys::*;
+
+    use crate::cpu::Core;
+
+    pub struct WatchdogConfig {
+        pub duration: core::time::Duration,
+        pub panic_on_trigger: bool,
+        // unused by now
+        pub cores: heapless::Vec<Core, 2>,
+    }
+
+    impl Default for WatchdogConfig {
+        fn default() -> Self {
+            let mut cores = heapless::Vec::new();
+            if CONFIG_ESP_TASK_WDT_CHECK_IDLE_TASK_CPU0 == 1 {
+                cores.push(Core::Core0).unwrap();
+            }
+            #[cfg(any(esp32, esp32s3))]
+            if CONFIG_ESP_TASK_WDT_CHECK_IDLE_TASK_CPU1 == 1 {
+                cores.push(Core::Core1).unwrap();
+            }
+            Self {
+                duration: core::time::Duration::from_secs(CONFIG_ESP_TASK_WDT_TIMEOUT_S as u64),
+                panic_on_trigger: false, // CONFIG_ESP_TASK_WDT_PANIC
+                cores,
+            }
+        }
+    }
+
+    #[cfg(esp_idf_version = "master")]
```

Comment:
If you are trying to branch by ESP IDF 4.X vs ESP IDF 5+, should be `not(esp_idf_version_major = "4")`

Reply:
Here is the problem that in idf-v5.0 the API not changed but the current master branch changed. So I think this could be v5.1 in the future.